File: decompiled/sources/com/hjq/http/model/CallProxy.java
package com.hjq.http.model;

import java.io.IOException;
import okhttp3.Call;
import okhttp3.Callback;
import okhttp3.Request;
import okhttp3.Response;
import okio.Timeout;

public final class CallProxy implements Call {
    private Call mCall;

    public CallProxy(Call call) {
        this.mCall = call;
    }

    public void setCall(Call call) {
        this.mCall = call;
    }

    public Request request() {
        Call call = this.mCall;
        if (call == null) {
            return null;
        }
        return call.request();
    }

    public Response execute() throws IOException {
        Call call = this.mCall;
        if (call == null) {
            return null;
        }
        return call.execute();
    }

    public void enqueue(Callback callback) {
        Call call = this.mCall;
        if (call != null) {
            call.enqueue(callback);
        }
    }

    public void cancel() {
        Call call = this.mCall;
        if (call != null) {
            call.cancel();
        }
    }

    public boolean isExecuted() {
        Call call = this.mCall;
        if (call == null) {
            return false;
        }
        return call.isExecuted();
    }

    public boolean isCanceled() {
        Call call = this.mCall;
        if (call == null) {
            return false;
        }
        return call.isCanceled();
    }

    public Timeout timeout() {
        Call call = this.mCall;
        if (call == null) {
            return null;
        }
        return call.timeout();
    }

    public Call clone() {
        Call call = this.mCall;
        if (call == null) {
            return null;
        }
        return call.clone();
    }
}

File: decompiled/sources/com/hjq/http/model/CallProxy.kt
package com.hjq.http.model

import java.io.IOException
import okhttp3.Call
import okhttp3.Callback
import okhttp3.Request
import okhttp3.Response
import okio.Timeout

class CallProxy(call: Call) : Call {
    private var mCall: Call = call

    fun setCall(call: Call) {
        mCall = call
    }

    override fun request(): Request {
        return mCall.request()
    }

    @Throws(IOException::class)
    override fun execute(): Response {
        return mCall.execute()
    }

    override fun enqueue(callback: Callback) {
        mCall.enqueue(callback)
    }

    override fun cancel() {
        mCall.cancel()
    }

    override fun isExecuted(): Boolean {
        return mCall.isExecuted
    }

    override fun isCanceled(): Boolean {
        return mCall.isCanceled
    }

    override fun timeout(): Timeout {
        return mCall.timeout()
    }

    public override fun clone(): Call {
        return mCall.clone()
    }
}
